funding/commitment_type_negotiation.go

```diff
@@ -360,6 +360,7 @@ func hasFeatures(local, remote *lnwire.FeatureVector,
 
 	for _, feature := range features {
 		if !local.HasFeature(feature) || !remote.HasFeature(feature) {
+			log.Infof("feature %d missing in local %t or in remote %t", feature, local.HasFeature(feature), remote.HasFeature(feature))
```

Comment:
commit title/body:

```
funding: add debug about missing feature bits

This commit adds a debug log in the funding package to communicate which 
feature (from either the local or remote node) is missing during feature 
negotiation.
```

Reply:
should this be a debug log instead?

Reply:
code should be wrapped at 80 chars

Reply:
I also dont think this is necessarily the correct spot for this log line. `hasFeatures` is a helper and a `false` return val doesnt necessarily mean something went wrong. 

I think we only really would want to log something if a miss-match in errors resulted in failed negotiation. In other words, perhaps we should instead add the missing feature to the error returned by `explicitNegotiateCommitmentType` so that when that error is logged, we see this extra information about what feature bit was missing.

Reply:
>I think we only really would want to log something if a miss-match in errors resulted in failed negotiation. In other words, perhaps we should instead add the missing feature to the error returned by explicitNegotiateCommitmentType so that when that error is logged, we see this extra information about what feature bit was missing.

This makes sense to me, anything that can help to catch this kind of bug just looking at the logs will be very helpful

Reply:
So a very rough idea would be something like this:

`hasFeatures` could be updated to return a set of missing feature bits. You probably want to actually return two separate sets here though: 1 for local missing & 1 for remote missing. Could also create a new function instead of changing `hasFeatures` so that `hasFeatures` can still do its thing as normal in places where we just want to know if there is a diff but dont want to error if so. 
```
// hasFeatures determines whether a set of features is supported by both the set
// of local and remote features. The set of missing feature bits is also
// returned.
func hasFeatures(local, remote *lnwire.FeatureVector,
	features ...lnwire.FeatureBit) (bool, []lnwire.FeatureBit) {

	var missing []lnwire.FeatureBit
	for _, feature := range features {
		if !local.HasFeature(feature) || !remote.HasFeature(feature) {
			missing = append(missing, feature)
		}
	}

	return len(missing) == 0, missing
}
```

Then you can update the cases in `explicitNegotiateCommitmentType` to something like this (note: i havent tested this, this is just an idea):

```
	case channelFeatures.OnlyContains(
		lnwire.ZeroConfRequired,
		lnwire.ScidAliasRequired,
		lnwire.ScriptEnforcedLeaseRequired,
		lnwire.AnchorsZeroFeeHtlcTxRequired,
		lnwire.StaticRemoteKeyRequired,
	):
		ok, missing := hasFeatures(
			local, remote,
			lnwire.ZeroConfOptional,
			lnwire.ScriptEnforcedLeaseOptional,
			lnwire.AnchorsZeroFeeHtlcTxOptional,
			lnwire.StaticRemoteKeyOptional,
		)
		if !ok {
			return 0, fmt.Errorf("%w: missing features: %v",
				errUnsupportedChannelType, missing)
		}

		return lnwallet.CommitmentTypeScriptEnforcedLease, nil
```

Reply:
This is a brilliant idea, thanks!

Reply:
@ellemouton do you allow me to mofidy the if check in this way? 😸 

```diff
-               if !hasFeatures(
+               if ok, missings := hasFeatures(
                        local, remote,
                        lnwire.ZeroConfOptional,
                        lnwire.AnchorsZeroFeeHtlcTxOptional,
                        lnwire.StaticRemoteKeyOptional,
-               ) {
+               ); !ok {

-                       return 0, errUnsupportedChannelType
+                       return 0, fmt.Errorf("%w: missing features: %v",
+                               errUnsupportedChannelType, missings)
                }
```